app/unit-tests/src/org/commcare/android/tests/queries/ExpressionCacheTest.java

```diff
@@ -0,0 +1,83 @@
+package org.commcare.android.tests.queries;
+
+import org.commcare.CommCareApplication;
+import org.commcare.android.CommCareTestRunner;
+import org.commcare.android.util.TestUtils;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.xpath.XPathParseTool;
+import org.javarosa.xpath.expr.XPathExpression;
+import org.javarosa.xpath.parser.XPathSyntaxException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.annotation.Config;
+
+/**
+ * Created by amstone326 on 2/2/18.
+ */
+
+@Config(application = CommCareApplication.class)
+@RunWith(CommCareTestRunner.class)
+public class ExpressionCacheTest {
+
+    private FormInstance mainFormInstance;
+
+    @Before
+    public void setup() {
+        mainFormInstance = new FormInstance(new TreeElement("data"));
+
+        TestUtils.initializeStaticTestStorage();
+        TestUtils.processResourceTransaction("/inputs/case_test_model_query_lookups.xml");
+    }
+
+    @Test
+    public void testFasterWithCaching() {
+        long timeWithoutCaching = time100Evaluations(false);
+        long timeWithCaching = time100Evaluations(true);
+        Assert.assertTrue("Evaluation time with caching should be shorter", timeWithCaching < timeWithoutCaching);
```

Comment:
Hm, if this difference is huge, I think that makes sense. I might push for simply counting the cache hit / miss ratio or something similar? Timing is such an awkward thing to test in some ways.

Reply:
@ctsims I added a new test below for testing the number of expression evaluations that used the cache, but the time differential here is actually very substantial. `timeWithoutCaching` is consistently almost twice as large as `timeWithCaching`. Given that, are you ok with leaving it?